subworkflows/local/bam_variant_calling_somatic_mutect2_ms/main.nf

```diff
@@ -0,0 +1,324 @@
+include { GATK4_MERGEVCFS                 as MERGE_MUTECT2                 } from '../../../modules/nf-core/gatk4/mergevcfs/main'
+include { GATK4_MERGEMUTECTSTATS          as MERGEMUTECTSTATS              } from '../../../modules/nf-core/gatk4/mergemutectstats/main'
+include { GATK4_LEARNREADORIENTATIONMODEL as LEARNREADORIENTATIONMODEL     } from '../../../modules/nf-core/gatk4/learnreadorientationmodel/main'
+include { GATK4_GATHERPILEUPSUMMARIES     as GATHERPILEUPSUMMARIES_NORMAL  } from '../../../modules/nf-core/gatk4/gatherpileupsummaries/main'
+include { GATK4_GATHERPILEUPSUMMARIES     as GATHERPILEUPSUMMARIES_TUMOR   } from '../../../modules/nf-core/gatk4/gatherpileupsummaries/main'
+include { GATK4_GETPILEUPSUMMARIES        as GETPILEUPSUMMARIES_NORMAL     } from '../../../modules/nf-core/gatk4/getpileupsummaries/main'
+include { GATK4_GETPILEUPSUMMARIES        as GETPILEUPSUMMARIES_TUMOR      } from '../../../modules/nf-core/gatk4/getpileupsummaries/main'
+include { GATK4_CALCULATECONTAMINATION    as CALCULATECONTAMINATION        } from '../../../modules/nf-core/gatk4/calculatecontamination/main'
+include { GATK4_FILTERMUTECTCALLS         as FILTERMUTECTCALLS             } from '../../../modules/nf-core/gatk4/filtermutectcalls/main'
+
+
+workflow BAM_VARIANT_CALLING_SOMATIC_MUTECT2_MS {
+    take:
+    input                     // channel: [ meta, [ input ], [ input_index ] ]
+    fasta                     // channel: /path/to/reference/fasta
+    fai                       // channel: /path/to/reference/fasta/index
+    dict                      // channel: /path/to/reference/fasta/dictionary
+    germline_resource         // channel: /path/to/germline/resource
+    germline_resource_tbi     // channel: /path/to/germline/index
+    panel_of_normals          // channel: /path/to/panel/of/normals
+    panel_of_normals_tbi      // channel: /path/to/panel/of/normals/index
+    intervals                 // channel: [mandatory] [ intervals, num_intervals ] or [ [], 0 ] if no intervals
+
+    main:
+    ch_versions = Channel.empty()
+    germline_resource_pileup     = germline_resource_tbi ? germline_resource : Channel.empty()
+    germline_resource_pileup_tbi = germline_resource_tbi ?: Channel.empty()
+
+
+    // Combine input and intervals for spread and gather strategy
+    input_intervals = input.combine(intervals)
+        // Move num_intervals to meta map and reorganize channel for MUTECT2_PAIRED module
+        .map{ meta, n_cram, n_crai, t_cram, t_crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], n_cram, n_crai, t_cram, t_crai, intervals ] }
+
+    // Prepare multi sample Mutect2 variant calling channels
+    // Separate normal cram files
+    ch_normal_cram = input_intervals.map{meta, n_cram, n_crai, t_cram, t_crai, intervals -> [[id: meta.patient, patient: meta.patient, normal_id: meta.normal_id, sex: meta.sex, num_intervals: meta.num_intervals], n_cram, n_crai]}.unique().groupTuple()
```

Comment:
You are right, this was because I branched from master and the input channel had intervals mixed in. We shouldn't need `unique` if we start with the cram inputs only but that needs some tweaking.

Reply:
Regarding the `size` parameter for `groupTuple`, it seems like there isn't an easy way to get the number of cram files per patient at the subworkflow level. This bottleneck happens upstream of the subwf [here](https://github.com/nf-core/sarek/blob/ed1cc8499366dcefea216fe37e36c6189537d57b/workflows/sarek.nf#L872) and I don't know how to overcome that.